vendor/github.com/docker/libnetwork/controller.go

```diff
@@ -343,6 +343,7 @@ func (c *controller) clusterAgentInit() {
 			c.agentClose()
 			c.cleanupServiceBindings("")
 
+			c.agentInitComplete()
```

Comment:
not super convinced by this, in reality on the node leave if the agent did not complete the activation this should not be released anyway. Is there a particular reason for this here?

Reply:
This is a temporary solution to resolve the `hang` issue only by testing.  I don`t know whether it is right, so seek community idea.

Reply:
if you remove it does it hangs still? if so can you send the SIGUSR1 and post the go routine stack trace?

Reply:
Docker may hang when running for a long time if remove it . So sorry, the test environment be destroyed.

Reply:
if you run it for a long time like with no swarm init/swarm leave happening?

Reply:
It can not be reproducted with no swarm init/swarm leave repeatedly.